File: leetcode/169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int> MaxsMap;

        for (auto num : nums) {
            MaxsMap[num]++;
        }

        int maximimum = 0;
        int majority = 0;
        for (const auto& pair : MaxsMap) {
            if (pair.second > maximimum) {
                maximimum = pair.second;
                majority = pair.first;
            }
        }
        return majority;
    } 
};

File: leetcode/169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        voting_count = 0
        current_candidate = None

        for num in nums:
            if voting_count == 0:
                current_candidate = num
            voting_count += (1 if num == current_candidate else -1)
        
        return current_candidate
        